vita3k/gui/src/game_context_menu.cpp

```diff
@@ -52,6 +55,63 @@ void delete_game(GuiState &gui, HostState &host, const std::string &title_id) {
     }
 }
 
+static std::vector<double> update_history_version;
+static std::vector<std::string> update_history_info;
+
+static bool get_update_history(GuiState &gui, HostState &host, const std::string &title_id) {
+    update_history_version.clear();
+    update_history_info.clear();
+
+    const auto change_info_path{ fs::path(host.pref_path) / "ux0/app" / title_id / "sce_sys/changeinfo/" };
+
+    std::string fname = fmt::format("changeinfo_{:0>2d}.xml", host.cfg.sys_lang);
+
+    if (!fs::exists(change_info_path / fname))
+        fname = "changeinfo.xml";
+
+    std::string xml_path = change_info_path.string() + fname;
+    pugi::xml_document doc;
+    pugi::xml_parse_result result = doc.load_file(xml_path.c_str());
+
+    for (const auto &update : doc.child("changeinfo")) {
+        update_history_version.push_back(update.attribute("app_ver").as_double());
+        update_history_info.push_back(update.text().as_string());
+    }
+
+    for (auto &i : update_history_info) {
+        const auto startpos = "<";
+        const auto endpos = ">";
+
+       if (i.find_first_of('\n') != std::string::npos)
+           i.erase(i.begin() + i.find_first_of('\n'));
+
+       while (i.find("</li> <li>") != std::string::npos)
+           if (i.find("</li> <li>") != std::string::npos)
+              i.replace(i.find("</li> <li>"), i.find(endpos) + 2 - i.find(startpos), "\n");
+       while (i.find("<li>") != std::string::npos)
+           if (i.find("<li>") != std::string::npos)
```

Comment:
Ditto.

Reply:
is same for all, i have learn how that works with learn hpo strip tag, and without while or if is broken